acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/privacy/PrivacyNode.java

```diff
@@ -12,111 +12,171 @@
-            Arrays.toString(
-                Arrays.stream(otherNodes).map(node -> node.orion.nodeUrl()).toArray())));
-    Enclave orionEnclave = new Enclave(orion.clientUrl());
+            Arrays.toString(otherNodes.stream().map(node -> node.pantheon.getName()).toArray()),
+            Arrays.toString(otherNodes.stream().map(node -> node.orion.nodeUrl()).toArray())));
+    Enclave enclaveClient = new Enclave(orion.clientUrl());
     SendRequest sendRequest1 =
         new SendRequestLegacy(
             "SGVsbG8sIFdvcmxkIQ==",
-            orion.getPublicKeys().get(0),
-            Arrays.stream(otherNodes)
-                .map(node -> node.orion.getPublicKeys().get(0))
+            orion.getDefaultPublicKey(),
+            otherNodes.stream()
+                .map(node -> node.orion.getDefaultPublicKey())
                 .collect(Collectors.toList()));
-    waitFor(() -> orionEnclave.send(sendRequest1));
+
+    Awaitility.await()
+        .until(
+            () -> {
+              try {
+                enclaveClient.send(sendRequest1);
+                return true;
+              } catch (final EnclaveException e) {
+                LOG.info("Waiting for enclave connectivity");
+                return false;
+              }
+            });
+  }
+
+  public void stop() {
+    pantheon.stop();
+    orion.stop();
+  }
+
+  @Override
+  public void close() {
+    pantheon.close();
+    orion.close();
+  }
+
+  public void start(final PantheonNodeRunner runner) {
+    orion.start();
+
+    final PrivacyParameters privacyParameters;
+    try {
+      privacyParameters =
+          new PrivacyParameters.Builder()
+              .setEnabled(true)
+              .setEnclaveUrl(orion.clientUrl())
+              .setEnclavePublicKeyUsingFile(orion.getConfig().publicKeys().get(0).toFile())
+              .setDataDir(Files.createTempDirectory("acctest-privacy"))
+              .setPrivateKeyPath(KeyPairUtil.getDefaultKeyFile(pantheon.homeDirectory()).toPath())
+              .build();
+    } catch (IOException e) {
+      throw new RuntimeException();
+    }
+    pantheon.setPrivacyParameters(privacyParameters);
+    pantheon.start(runner);
+  }
+
+  public void awaitPeerDiscovery(final Condition condition) {
+    pantheon.awaitPeerDiscovery(condition);
+  }
+
+  public String getName() {
+    return pantheon.getName();
+  }
+
+  public Address getAddress() {
+    return pantheon.getAddress();
+  }
+
+  public URI enodeUrl() {
+    return pantheon.enodeUrl();
+  }
+
+  public String getNodeId() {
+    return pantheon.getNodeId();
+  }
+
+  public <T> T execute(final Transaction<T> transaction) {
+    return pantheon.execute(transaction);
+  }
+
+  public void verify(final PrivateCondition expected) {
+    expected.verify(this);
+  }
+
+  public String getEnclaveKey() {
+    return orion.getDefaultPublicKey();
+  }
+
+  public String getTransactionSigningKey() {
+    return pantheon.keyPair().getPrivateKey().toString();
```

Comment:
It will always be the same key in acceptance tests. Agreed that this needs a bit more refactoring to make it more configurable.

Reply:
This is a little more complicated than a few lines fix. In the interest of time, I have made a JI for it for future work https://pegasys1.atlassian.net/browse/PAN-3113

I see it more of a "nice to have" rather than "must do right now"